File: core/src/main/kotlin/br/com/dillmann/nginxsidewheel/core/user/UserRepository.kt
package br.com.dillmann.nginxsidewheel.core.user

import br.com.dillmann.nginxsidewheel.core.common.pagination.Page
import java.util.UUID

interface UserRepository {
    suspend fun save(user: User)
    suspend fun deleteById(id: UUID)
    suspend fun findById(id: UUID): User?
    suspend fun findByUsername(username: String): User?
    suspend fun findPage(pageSize: Int, pageNumber: Int): Page<User>
    suspend fun count(): Long
}

File: core/user/user_repository.go
package user

import (
	"dillmann.com.br/nginx-ignition/core/common/pagination"
	"github.com/google/uuid"
)

type Repository interface {
	Save(user *User) error
	DeleteById(id uuid.UUID) error
	FindById(id uuid.UUID) (*User, error)
	FindByUsername(username string) (*User, error)
	FindPage(pageSize int64, pageNumber int64, searchTerms string) (*pagination.Page[User], error)
	IsEnabledById(id uuid.UUID) (bool, error)
	Count() (int64, error)
}
